src/js/view/floating/floating-controller.ts

```diff
@@ -0,0 +1,371 @@
+    }
+    setup(): void {
+        this._model.change('floating', this._boundInitFloatingBehavior);
+    }
+    initFloatingBehavior(): void {
+        // Don't reinitialize this behavior if the user dismissed the floating player
+        if (this._floatingStoppedForever) {
+            return;
+        }
+        // Setup floating scroll handler
+        viewsManager.removeScrollHandler(this._boundThrottledMobileFloatScrollHandler);
+        removeFPWatcher(this);
+        if (this.getFloatingConfig()) {
+            const fm = this.getFloatMode();
+            if (fm === 'notVisible') {
+                if (this._isMobile) {
+                    viewsManager.addScrollHandler(this._boundThrottledMobileFloatScrollHandler);
+                    this._boundThrottledMobileFloatScrollHandler();
+                } else {
+                    this.checkFloatIntersection();
+                }
+            } else if (fm === 'always') {
+                this.startFloating();
+            } else if (fm === 'never') {
+                this.stopFloating();
+            }
+        }
+    }
+    updatePlayerBounds(pb: BoundsObj): void {
+        this._playerBounds = pb;
+    }
+    getFloatingConfig(): FloatConfig | undefined {
+        return this._model.get('floating');
+    }
+    getFloatMode(): string {
+        const fc = this.getFloatingConfig();
+        return (fc && fc.mode) || 'notVisible';
+    }
+    resize(): void {
+        if (this._model.get('isFloating')) {
+            this.updateFloatingSize();
+        }
+    }
+    fosMobileBehavior(): boolean {
+        return this._isMobile && !deviceIsLandscape() && !this._model.get('fullscreen');
+    }
+    shouldFloatOnViewable(): boolean {
+        const state = this._model.get('state');
+        return state !== STATE_IDLE && state !== STATE_ERROR && state !== STATE_COMPLETE;
+    }
+    startFloating(mobileFloatIntoPlace?: boolean): void {
+        const playerBounds = this._playerBounds;
+        if (this.getFloatingPlayer() === null) {
+            this.setFloatingPlayer(this._playerEl);
+
+            this._model.set('isFloating', true);
+
+            addClass(this._playerEl, 'jw-flag-floating');
+
+            if (mobileFloatIntoPlace) {
+                // Creates a dynamic animation where the top of the current player
+                // Smoothly transitions into the expected floating space in the event
+                // we can't start floating at 62px
+                style(this._wrapperEl, {
+                    transform: `translateY(-${FLOATING_TOP_OFFSET - playerBounds.top}px)`
+                });
+
+                setTimeout(() => {
+                    style(this._wrapperEl, {
+                        transform: 'translateY(0)',
+                        transition: 'transform 150ms cubic-bezier(0, 0.25, 0.25, 1)'
+                    });
+                });
+            }
+
+            // Copy background from preview element, fallback to image config.
```

Comment:
Comment still says "fallback to image config". I'm guessing the idea here was if the player is idle, and the preview element was never populated... not sure how that would happen, but again it would not work with a platform config, only a basic shortcut `{ file, image }` setup which is more common in our test suites than anywhere IRL.

Reply:
Ah you are very right, removing the bad comment :+1: